File: services/networkx_service/src/process/network/envelope.py
import os


class Envelope:
    @staticmethod
    def send_algorithm_data(file_id, algorithm_name, algorithm_type):
        library_name = os.environ["LIBRARY_NAME"]

        return {
            "header": {
                "file_id": file_id,
                "algorithm_name": algorithm_name,
                "algorithm_type": algorithm_type,
                "library_name": library_name
            }
        }

    @staticmethod
    def send_user_data(socket_id, algorithm_name):
        library_name = os.environ["LIBRARY_NAME"]

        return {
            "header": {
                "socket_id": socket_id,
                "library_name": library_name,
                "algorithm_name": algorithm_name,
            }
        }

File: services/jgrapht_service/src/process/src/main/java/process/network/Envelope.java
package process.network;

import org.json.JSONObject;

public class Envelope {
    private static final String libraryName = System.getenv("LIBRARY_NAME");

    public static JSONObject sendAlgorithmData(String fileId, String algorithmName, String algorithmType) {
        return new JSONObject()
                .put("header",
                        new JSONObject()
                                .put("file_id", fileId)
                                .put("algorithm_name", algorithmName)
                                .put("algorithm_type", algorithmType)
                                .put("library_name", libraryName)
                );
    }

    public static JSONObject sendUserData(String socketId, String algorithmName) {
        return new JSONObject()
                .put("header",
                        new JSONObject()
                                .put("socket_id", socketId)
                                .put("library_name", libraryName)
                                .put("algorithm_name", algorithmName)
                );
    }
}
